### PR TITLE
Use paired end filenames to determine R1 and R2 for nextflow samplesheet

### PR DESCRIPTION
**Pull request by Bart-Edelbroek**
### The What

Determine R1 / R2 paired end read files in nextflow module

### The Why

When building the samplesheet, the first fastq file should be R1, and the second R2, or else the pipeline (especially rnaseq) will not run correctly

### The How

- Capture the 1 or 2 from [delim]R1[delim] / [delim]R2[delim] in the first regex capture group
  - If both R1 and R2 are found in one filename, raise an error (e.g. `myfastq_R2_r1.fastq.gz`)
- If nothing is found, try with [delim]1[delim] / [delim]2[delim]
  - e.g. would determine `myfastq_2_001.fq.gz` as R2
- Log which file was determined as which paired end read file

Note: This (intentionally) does not allow for single end sequencing, since we are running qd-rna with paired end reads (and accidentally running with a single fastq would be wrong). Let me know if we should allow single end sequencing. 

### This [update](https://semver.org/) is:
- [ ] **MAJOR** - when you make incompatible API changes
- [X] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### Tests

Tested with a manual run. Verified that the paired end read files were determined correctly. 

Depending on the filenames one can imagine edge cases which should / should not work.

### CODE REVIEW

**Comment by Bart-Edelbroek** [comment]
After internal discussions, I should maybe clarify that the R1/R2 order is not a problem when the data is available on seqstore (both files will be present), but mixups occur when using qd-rna to download the files using slims. From my testing it seems to do with the order in which the files are done downloading.

File: modules/nextflow/src/mixins.py [hunk 1]
```diff
@@ -1,9 +1,70 @@
...
+_NUMERIC  = re.compile(r"(?<=[._-])([12])(?=[._-])")
+
+def _infer_read(path: str | Path) -> tuple[int | None, str]:
+    """
+    Infer read number (1 or 2) from filename tokens.
+    Returns (read, how). If read is None, 'how' explains why.
+    """
+    name = Path(path).name
+
+    for pat, how in ((_EXPLICIT, "explicit R1/R2 token"), (_NUMERIC, "numeric 1/2 token")):
```
Comment (fannyhb) [inline modules/nextflow/src/mixins.py]: Can we have slithly more descriptive variables than pat and m? 
and some annotation of the code?

File: modules/nextflow/src/mixins.py [hunk 2]
```diff
@@ -1,9 +1,70 @@
...
+
+        for m in pat.finditer(name):
+            val = int(m.group(1))
+            seen.add(val)
+            last = val
+
+        if last is not None:
+            if len(seen) > 1:
+                return None, f"ambiguous: found both {sorted(seen)} via {how}"
+            return last, how
```
Comment (fannyhb) [inline modules/nextflow/src/mixins.py]: Forget my friday brain but is last supposed to be either a 1 or a 2?

**Review by fannyhb — APPROVED** [review]
Consider just changing one or two variable names to make it easier to understand the code.
Are you also pushing this to cellophane modules?

**Comment by Bart-Edelbroek** [comment]
I made some changes to clarify the code (I think). I also removed the 'last' variable because just taking the (only) value from the set should be fine

**Comment by Bart-Edelbroek** [comment]
We discussed also allowing running with a single fastq (single-end). I'll look into that before merging

**Comment by Bart-Edelbroek** [comment]
I tested with

```
- id: "55778845r"
  reads: 55054580
  files:
  - /oldseqstore/workspace/bart/qd_rna_rerun/scratch/batch3/251020-091648/unpack/55778845_240110_AHWV7HDRX3_S1_R2_001.fastq.gz
  - /oldseqstore/workspace/bart/qd_rna_rerun/scratch/batch3/251020-091648/unpack/55778845_240110_AHWV7HDRX3_S1_R1_001.fastq.gz

- id: "55778845x"
  reads: 55054580
  files:
  - /oldseqstore/workspace/bart/qd_rna_rerun/scratch/batch3/251020-091648/unpack/55778845_240110_AHWV7HDRX3_S1_R2_001.fastq.gz
```

In the log:

```
2026-01-30 10:29:11,884 : INFO : qlucore : Only one FASTQ file found for sample '55778845x', assumin
g single-end data.
2026-01-30 10:29:12,000 : DEBUG : rnafusion : Classified FASTQ /oldseqstore/workspace/bart/qd_rna_re
run/scratch/batch3/251020-091648/unpack/55778845_240110_AHWV7HDRX3_S1_R2_001.fastq.gz as R2 explicit
 R1/R2 token
2026-01-30 10:29:12,001 : DEBUG : rnafusion : Classified FASTQ /oldseqstore/workspace/bart/qd_rna_re
run/scratch/batch3/251020-091648/unpack/55778845_240110_AHWV7HDRX3_S1_R1_001.fastq.gz as R1 explicit
 R1/R2 token
```
